### ISSUE TITLE
Dotnet-podcasts app fails to build for iOS

### ISSUE THREAD

**Issue by LoopedBard3** [dotnet/maui#5400]
### Description

Trying to publish the dotnet-podcasts app (https://github.com/microsoft/dotnet-podcasts branch maui/perf) for iOS fails. The main errors seem to follow `ListenTogether/Emojis/EmojiPanel.razor : warning XA0101: @(Content) build action is not supported` and `Shared/Emojis/CryingFaceEmoji.razor : error : The path '../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/CryingFaceEmoji.razor' would result in a file outside of the app bundle and cannot be used`. In the docs, running dotnet build is mentioned but we need to generate the ipa files so we need to publish, although this seems to occur with both. This issue was found and seems like it may be related https://github.com/dotnet/maui/issues/791.

### Steps to Reproduce

General steps:
* Ensure a provisioning profile and signing stuff is available for publishing (if publishing) (not shown in mini script below)
* Install the latest dotnet sdk
* Install the latest Maui
* clone the dotnet-podcasts repository
* Integrate workaround for https://github.com/dotnet/sdk/issues/21877
* build the components per docs (https://github.com/microsoft/dotnet-podcasts/tree/main/src/Mobile)
* build/publish the main csproj for iOS arm-64

Script that should work on MAC:
```
curl -o NuGet.config 'https://raw.githubusercontent.com/dotnet/maui/main/NuGet.config'
curl -o dotnet-install.sh 'https://dotnet.microsoft.com/download/dotnet/scripts/v1/dotnet-install.sh'
chmod -R a+rx .
./dotnet-install.sh --channel 6.0.3xx --quality daily --install-dir .
./dotnet --info
./dotnet workload install maui --from-rollback-file https://aka.ms/dotnet/maui/main.json --configfile NuGet.config

git clone https://github.com/microsoft/dotnet-podcasts.git

cd dotnet-podcasts/src/Mobile
cp Microsoft.NetConf2021.Maui.csproj Microsoft.NetConf2021.Maui.csproj.bak
sed -i'' -e 's/net6.0-ios;net6.0-maccatalyst/net6.0-ios/g' Microsoft.NetConf2021.Maui.csproj
../../../dotnet build ../Web/Components/Podcast.Components.Maui.csproj
../../../dotnet build Microsoft.NetConf2021.Maui.csproj -bl:MauiPodcastiOS.binlog -f net6.0-ios --self-contained -r ios-arm64 -c Release /p:_RequireCodeSigning=false
cp Microsoft.NetConf2021.Maui.csproj.bak Microsoft.NetConf2021.Maui.csproj
```

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS

**Comment by eerhardt** [comment]
@Eilon @mkArtakMSFT @captainsafia - can someone take a look at this? This is blocking getting .NET Podcast app size under automation on iOS.

**Comment by eerhardt** [comment]
Here is a .binlog of trying to build the app on IOS:

[MauiPodcastiOS.zip](https://github.com/dotnet/maui/files/8328400/MauiPodcastiOS.zip)

There are 50 errors all coming from the `CollectBundleResources` task.

Given this, I think this might be an iOS tooling issue. @rolfbjarne @spouliot @dalexsoto - can someone on your team have a look?

<details>
  <summary>Click to see binlog</summary>

```
CollectBundleResources
    Assembly = /Users/eerhardt/git/maui/bin/dotnet/packs/Microsoft.iOS.Sdk/15.2.303-rc.1.55/tools/msbuild/iOS/../iOS/Xamarin.iOS.Tasks.dll
    Parameters
        ProjectDir = /Users/eerhardt/git/dotnet-podcasts/src/Web/Components
        BundleResources
            ListenTogether/Emojis/EmojiPanel.razor
                ExcludeFromSingleFile = true
            ListenTogether/Emojis/EmojiPanelButton.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogether.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogetherEmptyRoom.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogetherEpisode.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogetherInvite.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogetherJoinRoom.razor
                ExcludeFromSingleFile = true
            ListenTogether/ListenTogetherRoom.razor
                ExcludeFromSingleFile = true
            ListenTogether/UsernameForm.razor
                ExcludeFromSingleFile = true
            Player/FloatingPlayer.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/ClappingHandsEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/CryingFaceEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/FaceTearsOfJoyEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/HeartEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/ThumbsDownEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Emojis/ThumbsUpEmoji.razor
                ExcludeFromSingleFile = true
            Shared/Grid/Card.razor
                ExcludeFromSingleFile = true
            Shared/Grid/Grid.razor
                ExcludeFromSingleFile = true
            Shared/Modal.razor
                ExcludeFromSingleFile = true
            Shared/NoResults.razor
                ExcludeFromSingleFile = true
            Shared/Picture.razor
                ExcludeFromSingleFile = true
            Shared/PlayerBars.razor
                ExcludeFromSingleFile = true
            Shared/Spinner.razor
                ExcludeFromSingleFile = true
            Shared/SubtitlePage.razor
                ExcludeFromSingleFile = true
            Shared/Tags.razor
                ExcludeFromSingleFile = true
            Shared/TitlePage.razor
                ExcludeFromSingleFile = true
            Shared/ToggleSwitch.razor
                ExcludeFromSingleFile = true
            Shared/Tooltip.razor
                ExcludeFromSingleFile = true
            _Imports.razor
                ExcludeFromSingleFile = true
            wwwroot/audioJsInterop.js
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/button.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/fonts/SegoeUI.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/fonts/SegoeUIBold.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/fonts/SegoeUILight.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/fonts/SegoeUISemibold.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/fonts/SegoeUISemilight.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/icons/fonts/podcast-icons.svg
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/icons/fonts/podcast-icons.ttf
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/icons/fonts/podcast-icons.woff
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/icons/style.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/input.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/reset.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/site.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/splash.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/css/styles.css
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/images/bg-splash.jpg
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/images/bot-wait.svg
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/images/empty-results.png
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/images/logo-icon.svg
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
            wwwroot/themeJsInterop.js
                ExcludeFromSingleFile = true
                CopyToPublishDirectory = PreserveNewest
        ResourcePrefix = Resources
        OptimizePropertyLists = True
        OptimizePNGs = True
        TargetFrameworkMoniker = .NETCoreApp,Version=v6.0,Profile=ios
    Errors
        ListenTogether/Emojis/EmojiPanel.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/Emojis/EmojiPanel.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/Emojis/EmojiPanelButton.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/Emojis/EmojiPanelButton.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogether.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogether.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogetherEmptyRoom.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogetherEmptyRoom.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogetherEpisode.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogetherEpisode.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogetherInvite.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogetherInvite.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogetherJoinRoom.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogetherJoinRoom.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/ListenTogetherRoom.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/ListenTogetherRoom.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        ListenTogether/UsernameForm.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/ListenTogether/UsernameForm.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Player/FloatingPlayer.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Player/FloatingPlayer.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/ClappingHandsEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/ClappingHandsEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/CryingFaceEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/CryingFaceEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/FaceTearsOfJoyEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/FaceTearsOfJoyEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/HeartEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/HeartEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/ThumbsDownEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/ThumbsDownEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Emojis/ThumbsUpEmoji.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Emojis/ThumbsUpEmoji.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Grid/Card.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Grid/Card.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Grid/Grid.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Grid/Grid.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Modal.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Modal.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/NoResults.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/NoResults.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Picture.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Picture.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/PlayerBars.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/PlayerBars.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Spinner.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Spinner.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/SubtitlePage.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/SubtitlePage.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Tags.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Tags.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/TitlePage.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/TitlePage.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/ToggleSwitch.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/ToggleSwitch.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        Shared/Tooltip.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/Shared/Tooltip.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        _Imports.razor The path '../../../../../../../../dotnet-podcasts/src/Web/Components/_Imports.razor' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/audioJsInterop.js The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/audioJsInterop.js' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/button.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/button.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/fonts/SegoeUI.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/fonts/SegoeUI.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/fonts/SegoeUIBold.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/fonts/SegoeUIBold.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/fonts/SegoeUILight.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/fonts/SegoeUILight.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/fonts/SegoeUISemibold.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/fonts/SegoeUISemibold.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/fonts/SegoeUISemilight.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/fonts/SegoeUISemilight.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/icons/fonts/podcast-icons.svg The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/icons/fonts/podcast-icons.svg' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/icons/fonts/podcast-icons.ttf The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/icons/fonts/podcast-icons.ttf' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/icons/fonts/podcast-icons.woff The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/icons/fonts/podcast-icons.woff' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/icons/style.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/icons/style.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/input.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/input.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/reset.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/reset.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/site.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/site.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/splash.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/splash.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/css/styles.css The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/css/styles.css' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/images/bg-splash.jpg The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/images/bg-splash.jpg' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/images/bot-wait.svg The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/images/bot-wait.svg' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/images/empty-results.png The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/images/empty-results.png' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/images/logo-icon.svg The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/images/logo-icon.svg' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
        wwwroot/themeJsInterop.js The path '../../../../../../../../dotnet-podcasts/src/Web/Components/wwwroot/themeJsInterop.js' would result in a file outside of the app bundle and cannot be used.
         [/Users/eerhardt/git/dotnet-podcasts/src/Web/Components/Podcast.Components.Maui.csproj]
```
</details>

**Comment by mkArtakMSFT** [comment]
Thanks @eerhardt. That task is indeed defined in the MacDev.Tasks namespace, so they will have to look into this.

**Comment by Eilon** [comment]
@mkArtakMSFT this is likely a problem with how the BlazorWebView MAUI package combines the Razor SDK and iOS SDK, so it's most likely something @javiercn could help with.

**Comment by rolfbjarne** [comment]
I believe this is a duplicate of https://github.com/dotnet/maui/issues/791, which should have been fixed, but looks like the fix isn't fully fixed.

It seems this happens when building the `Podcast.Components.Maui.csproj` project, and for some reason the `ConvertStaticWebAssetsToMauiAssets` target (which applies the fix above), isn't executed. I'm guessing this is because the `Podcast.Components.Maui.csproj` project doesn't set the `UseMaui` property to `true`, and thus the maui targets aren't loaded.

**Comment by eerhardt** [comment]
> I'm guessing this is because the Podcast.Components.Maui.csproj project doesn't set the UseMaui property to true

It does set `UseMaui=true`:
https://github.com/microsoft/dotnet-podcasts/blob/b8b76978d87222fabeee22f732ae352244431e4c/src/Mobile/Microsoft.NetConf2021.Maui.csproj#L10

**Comment by Eilon** [comment]
BTW for some projects in that app there is more than 1 CSPROJ, and which one is used depends on which project it's being referenced from.

**Comment by eerhardt** [comment]
Setting `UseMaui=true` in the Podcast.Components.Maui.csproj fixed it for me locally.

Is this the recommended fix? It seems odd that Android doesn't need this, but iOS does?

**Comment by rolfbjarne** [comment]
> Because that's supposed to be broken.

Not sure I understand why is it supposed to be broken?

> Maybe the app compiles but it doesn't run?

`Podcast.Components.Maui.csproj` is a library project, so no, it doesn't run.

**Comment by Eilon** [comment]
@rolfbjarne there's this long-standing bug: https://github.com/dotnet/maui/issues/3288

Sure the library doesn't run, but I wonder if an app using that library would actually work, or if it's missing content due to some build problem.

**Comment by Eilon** [comment]
@eerhardt I'll see if I can get my Mac updated to a working state (I haven't run MAUI on it recently) and let you know if I can try it.

**Comment by javiercn** [comment]
@Eilon @LoopedBard3 I think the issue is in the way the app is defined.

Something in [here](https://github.com/microsoft/dotnet-podcasts/blob/main/src/Web/Components/Podcast.Components.Maui.csproj) seems to be bringing in the Maui targets into the Razor Class Library. It's not clear to me why this is not "just" a net6.0 library, but that's what I would expect.

The issue happens because Maui targets are added to the project (not sure why) and there are no targets (like the ones we use in our Blazor webview project) to adjust the outcomes of some of those targets.

My suggestion would be to make that library a plain Razor Class Library targeting .NET 6.0

**Comment by eerhardt** [comment]
Thanks @javiercn for the analysis! I was able to unblock the .NET Podcasts app by switching the library to target net6.0.

I'm going to close this issue for now, as there is no change needed in dotnet/maui.
